File: app/src/main/java/com/softaai/designpatterns/java/structural/bridge/BridgeDesignPatternDemo.java
package com.softaai.designpatterns.java.structural.bridge;

/**
 * Created by Amol Pawar on 03-04-2019.
 * softAai Apps
 */
public class BridgeDesignPatternDemo {

    public static void main(String[] args){
        Shape redCircle = new Circle(100,100, 10, new RedCircle());
        Shape greenCircle = new Circle(100,100,10,new GreenCircle());
        redCircle.draw();
        greenCircle.draw();
    }
}

File: app/src/main/java/com/softaai/designpatterns/kotlin/structural/bridge/BridgeDesignPatternDemo.kt
package com.softaai.designpatterns.kotlin.structural.bridge


/**
 * Created by Amol Pawar on 03-04-2019.
 * softAai Apps
 */
object BridgeDesignPatternDemo {
    @JvmStatic
    fun main(args : Array<String>){
        val redCircle = Circle(100,100, 10, RedCircle())
        val greenCircle = Circle(100,100,10, GreenCircle())
        redCircle.draw()
        greenCircle.draw()
    }
}